client/adjudicate.go

```diff
@@ -200,9 +203,12 @@ func (c *Channel) ForceUpdate(ctx context.Context, update func(*channel.State))
 	ar := c.machine.AdjudicatorReq()
 
 	// Update state
-	state := c.machine.State().Clone()
+	state := ar.Tx.State.Clone()
```

Comment:
It was a bit hard for me to understand why `ar.Tx.State.Clone()` is used over `c.machine.State().Clone()`. Only after looking into `c.machine,AdjudicatorReq()`, I understood they were the same.

Because, we are locking the machine mutex and the `currentTx` cannot change,
I prefer we use the previous version ``c.machine.State().Clone()`. and move the definition of `ar`, just before the line where it is required.
Then the function would read as
1...... 
2. Fetch the latest state and update it.
3. .... 
4. Generate adjudicator request and make on-chain call.

Reply:
That doesn't work because the adjudicator request must contain the previous state.

Reply:
Okay. Now, I understand why this change makes sense. And I read it as
1. Generate adjudicator request from the current state.
2. Update the state in the adjudicator request.
3. Send both the old and the new states to the contract.